src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -178,6 +178,34 @@ public async Task<IEnumerable<PackageRestoreData>> GetPackagesInSolutionAsync(st
             return packageReferencesDict;
         }
 
+        private async Task<Dictionary<string, RestoreAuditProperties>> GetRestoreAuditProperties()
+        {
+            var restoreAuditProperties = new Dictionary<string, RestoreAuditProperties>(PathUtility.GetStringComparerBasedOnOS());
+            if (!await SolutionManager.IsSolutionAvailableAsync())
+            {
+                return restoreAuditProperties;
+            }
+
+            foreach (var nuGetProject in (await SolutionManager.GetNuGetProjectsAsync()))
```

Comment:
please save the await in a variable. Since NuGet has an overreliance on `IEnumerable`, I bet we should also use `.NoAllocEnumerate()` as well, but I'm not motivated to load this PR in an IDE jus to use IntelliSense to check.